src/app.module.ts

```diff
@@ -34,47 +34,53 @@ require('dotenv').config(); // eslint-disable-line
 @Module({
   imports: [
     ScheduleModule.forRoot(),
+    GCloudSecretManagerModule.withConfig(process.env.PARENT),
     TypeOrmModule.forRootAsync({
-      imports: [GCloudSecretManagerModule.withConfig(process.env.PARENT)],
       inject: [GCloudSecretManagerService],
       useFactory: async (
         gCloudSecretManagerService: GCloudSecretManagerService,
       ) => {
-        await gCloudSecretManagerService.loadSecrets();
         return {
           type: 'postgres',
-          host: process.env.HOST_POSTGRES,
+          host: gCloudSecretManagerService
+            .getSecret('POSTGRES_HOST')
+            .toString(),
           port: 5432,
           username: gCloudSecretManagerService
             .getSecret('POSTGRES_USERNAME')
             .toString(),
           password: gCloudSecretManagerService
             .getSecret('POSTGRES_PASSWORD')
             .toString(),
-          database: process.env.DB_POSTGRES,
+          database: gCloudSecretManagerService
+            .getSecret('POSTGRES_DB')
+            .toString(),
           entities: [LabRating, TransactionRequest],
           autoLoadEntities: true,
         };
       },
     }),
     TypeOrmModule.forRootAsync({
-      imports: [GCloudSecretManagerModule.withConfig(process.env.PARENT)],
+      name: 'dbLocation',
       inject: [GCloudSecretManagerService],
       useFactory: async (
         gCloudSecretManagerService: GCloudSecretManagerService,
       ) => {
-        await gCloudSecretManagerService.loadSecrets();
         return {
           type: 'postgres',
-          host: process.env.HOST_POSTGRES,
+          host: gCloudSecretManagerService
+            .getSecret('POSTGRES_HOST')
+            .toString(),
           port: 5432,
           username: gCloudSecretManagerService
             .getSecret('POSTGRES_USERNAME')
             .toString(),
           password: gCloudSecretManagerService
             .getSecret('POSTGRES_PASSWORD')
             .toString(),
-          database: process.env.DB_LOCATION,
+          database: gCloudSecretManagerService
+            .getSecret('POSTGRES_DB_LOCATIONS')
+            .toString(),
```

Comment:
Not everything needs to be kept inside Google Secret Manager. Please undo the code samples where you removed `process.env` @rubenkristian.

Reply:
oke.